### PR TITLE
Fix/tscompilererrors

### PR DESCRIPTION
**Pull request by nisteuve**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Are the files formatted correctly?

## Description

There were typescript compiler errors due to two factors:

-Use of "const enum" rather than "enum"
-Missing return type for several exported functions

## Related PRs or issues

https://github.com/microsoft/appcenter-sdk-react-native/issues/749

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,10 +2,21 @@
 
 ## Version 2.5.1 (Under development)
 
+### App Center
+
+* **[Fix]** Fix typescript compiler errors related to const enum usage and missing return types
```
Comment (guperrot) [inline CHANGELOG.md]:
```suggestion
* **[Fix]** Fix typescript compiler errors related to const enum usage and missing return types.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,10 +2,21 @@
...
+
 ### App Center Analytics
 
 * **[Fix]** Avoid throwing an error when `trackEvent` is called with null property values.
 
+### App Center Crashes
+
+* **[Fix]** Fix typescript compiler errors related to const enum usage
```
Comment (guperrot) [inline CHANGELOG.md]:
```suggestion
* **[Fix]** Fix typescript compiler errors related to const enum usage.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,10 +2,21 @@
...
+
+### App Center Data
+
+* **[Fix]** Fix typescript compiler errors related to const enum usage
```
Comment (guperrot) [inline CHANGELOG.md]:
```suggestion
* **[Fix]** Fix typescript compiler errors related to const enum usage.
```

File: appcenter-data/__tests__/Data.js [hunk 4]
```diff
@@ -77,12 +77,13 @@ describe('App Center Data read operation tests', () => {
...
 
 describe('App Center Data list operation tests', () => {
-  test('list is called ', async () => {
+  test('list is called with partition and TimeToLive.DEFAULT', async () => {
     const partition = Data.DefaultPartitions.USER_DOCUMENTS;
     const spy = jest.spyOn(NativeModules.AppCenterReactNativeData, 'list');
     const result = await Data.list(partition);
     console.log(result);
-    expect(spy).toHaveBeenCalledWith(partition);
+    const readOptions = { "timeToLive": Data.TimeToLive.DEFAULT };
```
Comment (guperrot) [inline appcenter-data/__tests__/Data.js]: undefined should map to default, we are supposed to have code to handle the default case
if you had to update the unit test then it might hide a real bug in the SDK

Comment (dhei) [reply appcenter-data/__tests__/Data.js]: @guperrot `await Data.list(partition)` is called without TTL parameter (line 82) and the SDK correctly treats it `TimeToLive.DEFAULT`. The problem was this test case was failing on develop branch but unit tests of Data module and Auth module are not executed in the CI. So this PR fixed a bug in the unit test, and now all the JS unit tests are executed in the CI.

File: appcenter-data/__tests__/Data.js [hunk 5]
```diff
@@ -77,12 +77,13 @@ describe('App Center Data read operation tests', () => {
 });
 
 describe('App Center Data list operation tests', () => {
-  test('list is called ', async () => {
+  test('list is called with partition and TimeToLive.DEFAULT', async () => {
```
Comment (dhei) [inline appcenter-data/__tests__/Data.js]: As mentioned in https://github.com/microsoft/appcenter-sdk-react-native/pull/767/files#r344940943, there is a gap in the test coverage. `Data.list` API takes partition and readOptions as parameters but we don't have a test case covering list API called with both parameters.

**Comment by guperrot** [comment]
Build error is

```
/Users/runner/runners/2.160.0/work/1/s/appcenter-data/__tests__/Data.js
  85:27  error  Unnecessarily quoted property 'timeToLive' found  quote-props
  85:27  error  Strings must use singlequote                      quotes
```

File: appcenter-data/__tests__/Data.js [hunk 6]
```diff
@@ -77,12 +77,13 @@ describe('App Center Data read operation tests', () => {
...
     const partition = Data.DefaultPartitions.USER_DOCUMENTS;
     const spy = jest.spyOn(NativeModules.AppCenterReactNativeData, 'list');
     const result = await Data.list(partition);
     console.log(result);
-    expect(spy).toHaveBeenCalledWith(partition);
+    const readOptions = { 'timeToLive': Data.TimeToLive.DEFAULT };
```
Comment (dhei) [inline appcenter-data/__tests__/Data.js]:
```
/Users/runner/runners/2.160.0/work/1/s/appcenter-data/__tests__/Data.js
  85:27  error  Unnecessarily quoted property 'timeToLive' found  quote-props
```

**Comment by MatkovIvan** [comment]
@dhei It's too late, but

I don't think that changing `const enum` to `enum` is the right solution. I placed some references in the comment to work item a while ago (when I moved it to triage), but it looks like it was missed some how.

See https://lukasbehal.com/2017-05-22-enums-in-declaration-files/
**TLDR**: Do not define your enums in your TypeScript Declaration (`*.d.ts`) files, ever.
A union type of literals should be used instead.
